editor.planx.uk/src/pages/Preview/Node.tsx

```diff
@@ -75,16 +75,13 @@ interface Props {
 }
 
 const Node: React.FC<Props> = (props) => {
-  const [childNodesOf, isFinalCard, resetPreview, cachedBreadcrumbs] = useStore(
-    (state) => [
-      state.childNodesOf,
-      state.isFinalCard(),
-      state.resetPreview,
-      state.cachedBreadcrumbs,
-    ],
-  );
-
-  const handleSubmit = isFinalCard ? undefined : props.handleSubmit;
+  const [childNodesOf, resetPreview, cachedBreadcrumbs] = useStore((state) => [
+    state.childNodesOf,
+    state.resetPreview,
+    state.cachedBreadcrumbs,
+  ]);
+
+  const handleSubmit = props.handleSubmit;
```

Comment:
Ideally, in the near future, we can/will re-introduce `isFinalCard` at the `Preview/Node` level, but we'll want to do more thorough performance reviews and add test coverage before doing this - so have adjusted this to keep current behavior in the meantime and instead _only_ reference `isFinalCard` within the Confirmation component itself rather than _any_ component

Reply:
Totally agree - this is the right approach here 👍